Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control._graphClient != null && !string.IsNullOrWhiteSpace(control.DriveUrl))
+            {
+                if (Uri.IsWellFormedUriString(control.DriveUrl, UriKind.Absolute))
+                {
+                    await control.InitDrive(control.DriveUrl);
+                }
+            }
+        }
+
+        private static void DetailPanePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            SharePointFiles control = d as SharePointFiles;
+            if (control.IsDetailPaneVisible)
+            {
+                control.ShowDetailsPane();
+            }
+        }
+
+        /// <summary>
+        /// Called when applying the control template.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            _list = GetTemplateChild("list") as ListView;
+            _back = GetTemplateChild("back") as Button;
+            _upload = GetTemplateChild("upload") as Button;
+            _share = GetTemplateChild("share") as Button;
+            _download = GetTemplateChild("download") as Button;
+            _delete = GetTemplateChild("delete") as Button;
+            _error = GetTemplateChild("error") as HyperlinkButton;
+            _cancel = GetTemplateChild("cancel") as Button;
+            _hasMore = GetTemplateChild("hasMore") as Button;
+            _thumbnail = GetTemplateChild("thumbnail") as Grid;
+            _thumbnailImage = GetTemplateChild("thumbnailImage") as Windows.UI.Xaml.Controls.Image;
+            _details = GetTemplateChild("details") as ScrollViewer;
+            _status = GetTemplateChild("status") as TextBlock;
+            _list.SelectionChanged += List_SelectionChanged;
+            _list.ItemClick += List_ItemClick;
+            _back.Click += Back_Click;
+            _upload.Click += Upload_Click;
+            _share.Click += Share_Click;
+            _download.Click += Download_Click;
+            _delete.Click += Delete_Click;
+            _error.Click += Error_Click;
+            _cancel.Click += Cancel_Click;
+            _hasMore.Click += LoadMore_Click;
+
+            base.OnApplyTemplate();
+        }
+
+        private async void List_ItemClick(object sender, ItemClickEventArgs e)
+        {
+            if (e.ClickedItem is DriveItem driveItem && driveItem.Folder != null)
+            {
+                _driveItemPath.Push(driveItem.Id);
+                _back.Visibility = Visibility.Visible;
+                await LoadFiles(driveItem.Id);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves an appropriate Drive URL from a SharePoint document library root URL
+        /// </summary>
+        /// <param name="rawDocLibUrl">Raw URL for SharePoint document library</param>
+        /// <returns>Drive URL</returns>
+        public async Task<string> GetDriveUrlFromSharePointUrl(string rawDocLibUrl)
```

Comment:
Rename by `GetDriveUrlFromSharePointUrlAsync` for consistency

Reply:
@Odonno done